src/benchmarks/echo-c/main.c

```diff
@@ -28,19 +28,19 @@ int main(void)
     size_t n = 0;
 
     while (1) {
-        nread = fread(buffer + n, 1, MAX_REQUEST_SIZE - n, stdin);
+        nread = fread(buffer, 1, MAX_REQUEST_SIZE, stdin);
         if (nread < 0) {
             break; // Error encountered.
         } else if (nread == 0) {
             break; // End of file reached.
         } else {
             n += nread; // Read some bytes.
         }
-    }
 
-    if (n > 0) {
-        fwrite(buffer, 1, n, stdout);
-        fflush(stdout);
+        if (n > 0) {
+            fwrite(buffer, 1, n, stdout);
+            fflush(stdout);
+        }
     }
 
     return 0;
```

Comment:
We dont need two variables `n` and `nread` anymore. Could you change the benchmark to use just one?